internal_ws/ykshim/src/lib.rs

```diff
@@ -3,9 +3,124 @@
 //! For more information, see this section in the documentation:
 //! https://softdevteam.github.io/ykdocs/tech/yk_structure.html
 
-/// The production API.
-mod prod_api;
-
-/// The testing API.
-/// These functions are only exposed to allow testing from the external workspace.
 mod test_api;
+
+use std::ffi::{c_void, CString};
+use std::os::raw::c_char;
+use ykcompile::CompiledTrace;
+use yktrace::{
+    sir::{SirTrace, SIR},
+    tir::TirTrace,
+    ThreadTracer,
+};
+
+/// The different ways by which we can collect a trace.
+#[derive(Clone, Copy)]
+#[repr(u8)]
+#[allow(dead_code)]
+enum TracingKind {
+    /// Software tracing via ykrustc.
+    SoftwareTracing = 0,
```

Comment:
Do we need to hardcode these numbers? [I appreciate this is just moved code, but I might as well ask while I'm looking at it!]

Reply:
It's due to the workspace model. The numbers have to be duplicated in both workspaces whilst also being stable across the C ABI. I don't see any value in adding constants here, as the enum already serves to define constants.

Reply:
OK.